.travis.yml

```diff
@@ -1,58 +1,75 @@
 language: rust
 sudo: false
+rust: nightly
 
 matrix:
   include:
     - name: "miri"
-      rust: nightly
-      install:
-        - rustup component add rust-src
-        - cargo +nightly install --force --git https://github.com/rust-lang/miri miri
-        - cargo install xargo
-      script:
-        - cargo clean
-        - cargo +nightly miri test
-    - rust: 1.29.0
-    - rust: stable
-    - rust: beta
-    - rust: nightly
-    - rust: nightly
-      env:
-       - FEATURES='nightly'
+      env: TARGET=x86_64-unknown-linux-gnu
+      script: sh ci/miri.sh
+    - name: "rustfmt/clippy"
+      env: TARGET=i586-unknown-linux-gnu
+      script: sh ci/tools.sh
+    - name: "docs"
+      env: TARGET=x86_64-unknown-linux-gnu
+      script: cargo -vv doc --features nightly,serde,rayon
+      deploy:
+        provider: pages
+        skip-cleanup: true
+        github-token: $GITHUB_TOKEN
+        local-dir: target/doc
+        keep-history: false
+        on:
+          branch: master
```

Comment:
I don't know if you are interested into deploying the docs with the nightly features to github-pages. If not I can just remove the `deploy` key. Otherwise, you'd need to generate a token and push it to travis as `GITHUB_TOKEN`.

Reply:
I think the only difference is that `HashMap::new` is const fn on nightly. Not worth a completely separate set of docs IMO.

Reply:
I thought the serde and rayon features would add more stuff. I'll remove the deploy key then.

Reply:
Oh! I forgot about those! Yes, please do keep the docs in that case.